lib/matplotlib/cbook.py

```diff
@@ -2661,3 +2661,210 @@ def __exit__(self, exc_type, exc_value, traceback):
                     os.rmdir(path)
                 except OSError:
                     pass
+
+
+class _FuncInfo(object):
+    """
+    Class used to store a function
+
+    Each object has:
+    * The direct function (function)
+    * The inverse function (inverse)
+    * A boolean indicating whether the function
+      is bounded in the interval 0-1 (bounded_0_1), or
+      a method that returns the information depending
+      on this
+    * A callable (check_params) that takes a list of the parameters
+      and returns a boolean specifying if a certain combination of
+      parameters is valid. It is only required if the function as
+      parameters and some of them are restricted.
+
+    """
+    def __init__(self, function, inverse, bounded_0_1=True, check_params=None):
+        self.function = function
+        self.inverse = inverse
+
+        if callable(bounded_0_1):
+            self._bounded_0_1 = bounded_0_1
+        else:
+            self._bounded_0_1 = lambda x: bounded_0_1
+
+        if check_params is None:
+            self._check_params = lambda x: True
+        elif callable(check_params):
+            self._check_params = check_params
+        else:
+            raise ValueError("Check params must be a callable taking a list "
+                             "with function parameters and returning a "
+                             "boolean indicating whether that combination of "
+                             "parameters is valid. In case of validity for "
```

Comment:
In that case, maybe just "Invalid 'check_params' argument"? Because 'None' is also a valid value, and that way it does not necessarily implies that it has to be a function.